### ISSUE TITLE
libpmempool rm: if file is not pool or poolset it should not be deleted

### ISSUE THREAD

**Issue by szychows** [pmem/issues#435]
Consider following text file "textfile.txt":
`"some text"`
and then using
`pmempool_rm("textfile.txt", 0) `
will delete the file. pmempool_rm() should check if it is file or poolset before attempting any actions.

Found on: 1.2+wtp1-227-gac3524e

**Comment by wojtuss** [comment]
A damaged part can also be considered neither a poolset file nor a file with a pool and we still need a means of removing the part, particularly in the case of device DAX where simple rm won't do.

**Comment by plebioda** [comment]
@wojtuss Yes you are right, in such case you can use pmempool_rm(path, PMEMPOOL_RM_FORCE).

**Comment by krzycz** [comment]
What are we going to do with this issue? Can we close it as "not a bug"?

**Comment by krzycz** [comment]
Closing as "not a bug".

**Comment by kamdiedrich** [comment]
I think it should be fixed.
What will happen if we pass to pmempool rm some file that we don't want to remove because it isn't poolset file or pool file?

**Comment by krzycz** [comment]
The `pmempool rm` command is just a a convenience wrapper around `rm` or `unlink`. If you make a mistake (i.e. `rm -rf $HOME`), sorry...
The only special cases it handles is when the input file is a pool set file (then it removes all the pool set parts / replicas / etc.), or when it points to Device DAX (then device is zeroed instead of being removed).
It does not validate whether the pool file, part files or DAX devices do really belong to the pool and whether they contain the valid data (i.e. pool headers). Moreover, once we introduced **NOHDRS** option to the pool sets, it may happen we're not able to determine if given file is rally a part of the pool set or not.

If you think it's worth to have some additional checks, I suggest to change this issue into a feature request, not a bug.

**Comment by pbalcer** [comment]
I this we should close this.
@marcinslusarz @dawidpaluszkiewicz

**Comment by marcinslusarz** [comment]
I agree with Kamil. I think pmempool_rm should verify that it knows what it's unlinking. So if file doesn't look like poolset or part with a header it should refuse to do anything. For NOHDRS/single header case and Device DAX with uninitialized header there is still an option to say "I don't care, remove it anyway".
